include/deal.II/fe/mapping_fe.h

```diff
@@ -0,0 +1,741 @@
+ * In particular we think of a collections of a FE_Q or
+ * Bezier finite element (FE_Bernstein) repeated a number of times equal to the space
+ * dimension. The idea is to construct the mapping using a vector of control
+ * points, a DoFHandler associated to the geometry of the problem and a
+ * ComponentMask that tells us which components to use for the mapping.
+ * This mapping will grab from the DoFHandler the finite element, or better
+ * the collection of finite elements, to compute the mapping shape functions.
+ * So we will have two different Finite Element and DoFHandler, one for the
+ * solution field and one to describe the geometry of the problem. Historically
+ * in the deal.II library there was not such a distinction. The differences
+ * between this mapping and the MappingQ class are quite important.
+ * The MappingFE, being a generalization, requires a higher level of abstraction.
+ * This is the reason why it takes a DoFHandler and a vector of control points
+ * that are the coefficients of the shape function (so in general it is a vector
+ * of coefficient).
+ *
+ *
+ * Typically, the DoFHandler operates on a finite element that is constructed
+ * as a system element (FESystem) from continuous FE_Q() objects. An example
+ * is shown below:
+ * @code
+ *    const FE_Q<dim,spacedim> feq(1);
+ *    const FESystem<dim,spacedim> fesystem(feq, spacedim);
+ *    DoFHandler<dim,spacedim> dhq(triangulation);
+ *    dhq.distribute_dofs(fesystem);
+ *    Vector<double> eulerq(dhq.n_dofs());
+ *    const ComponentMask mask(spacedim, true);
+ *    MappingFE<dim,spacedim> map(eulerq, dhq, mask);
+ *    map.update_euler_vector_using_triangulation(eulerq);
+ * @endcode
+
+
+ *
+ * @author Luca Heltai, Marco Tezzele 2013, 2015
+ */
+template <int dim, int spacedim=dim,
+          class DH=DoFHandler<dim,spacedim>,
+          class VECTOR=Vector<double> >
+class MappingFE : public Mapping<dim,spacedim>
+{
+public:
+  /**
+   * Constructor. The first argument is a VECTOR that specifies the
+   * transformation of the domain from the reference to the current
+   * configuration. This is filled calling the method
+   * update_euler_vector_using_triangulation.
+   */
+  MappingFE (const VECTOR  &euler_vector,
+             const DH      &euler_dof_handler,
+             const ComponentMask mask=ComponentMask());
+
+  /**
+   * Copy constructor. Performs a deep copy, i.e. duplicates what #tensor_pols
+   * points to instead of simply copying the #tensor_pols pointer as done by a
+   * default copy constructor.
+   */
+  MappingFE (const MappingFE<dim,spacedim,DH,VECTOR> &mapping);
+
+  /**
+   * Destructor.
+   */
+  virtual ~MappingFE ();
+
+  /** Fill the euler vector with
+  the information coming from
+  the triangulation. Makes this
+  map equivalent to MappingQ1,
+  and it works ONLY if the
+  underlying fe has support
+  points. */
+  void update_euler_vector_using_triangulation(VECTOR &vector);
```

Comment:
I don't understand what this function does, based on the documentation. This may be addressed by augmenting the class documentation.

(Btw, should the reference be `const`? If not, then that points to another issue I don't understand.)